src/components/TimelineComponent.tsx

```diff
@@ -762,7 +838,7 @@ export default class TimelineComponent
         });
 
     // finds starting index
-    dataIdx = Math.floor(-deltaX / (scale * barWidth));
+    dataIdx = Math.floor(-deltaX / (this.scale * barWidth));
     data = csvData.slice(dataIdx, numBars + dataIdx);
     ordinals = data.map((d: any) => d[xColumn]);
 
```

Comment:
overall good grouping of logic. Could maybe break down some longer code and add some comments but overall good work!